sql/src/main/java/io/crate/execution/engine/pipeline/ProjectionToProjectorVisitor.java

```diff
@@ -591,16 +591,19 @@ public Projector visitWindowAgg(WindowAggProjection windowAgg, Context context)
             ctx.add(functionAndInputsEntry.getValue());
 
             FunctionImplementation impl = this.functions.getQualified(functionAndInputsEntry.getKey().info().ident());
-            assert impl instanceof AggregationFunction : "We currently only support aggregation functions as window functions";
-            windowFunctions.add(
-                new AggregateToWindowFunctionAdapter(
-                    ctx.topLevelInputs().toArray(new Input[0]),
-                    (AggregationFunction) impl,
-                    ctx.expressions(),
-                    indexVersionCreated,
-                    bigArrays,
-                    context.ramAccountingContext)
-            );
+            if (impl instanceof AggregationFunction) {
+                windowFunctions.add(
+                    new AggregateToWindowFunctionAdapter(
+                        ctx.topLevelInputs().toArray(new Input[0]),
+                        (AggregationFunction) impl,
+                        ctx.expressions(),
+                        indexVersionCreated,
+                        bigArrays,
+                        context.ramAccountingContext)
+                );
+            } else if (impl instanceof WindowFunction) {
+                windowFunctions.add((WindowFunction) impl);
+            }
```

Comment:
please add a `else` which raises an assertion error that there must be a WindowFunction or an AggregationFunction. It is better to be defensive than to later on have to debug some weird behaviour.